M/international/locations/states

RaimondasSapola opened the pull request:
<!--
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
 -->

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop / 1.7.6.x / 1.6.1.x
| Description?  | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc.
| Type?         | bug fix / improvement / new feature / refacto / critical
| Category?     | FO / BO / CO / IN / WS / TE
| BC breaks?    | Does it break backward compatibility? yes/no
| Deprecations? | Does it deprecate an existing feature? yes/no
| Fixed ticket? | (optional) If this PR fixes an [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Please indicate how to best verify that this PR is correct.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

RaimondasSapola committed:
States list and filters migration

RaimondasSapola committed:
State page assets

RaimondasSapola committed:
Asset update

RaimondasSapola committed:
State page actions

rokaszygmantas commented on admin-dev/themes/new-theme/js/pages/state/index.js:
```diff
@@ -0,0 +1,64 @@
+
+  new FormSubmitButton();
+
+  new TaggableField({
```
where do you use taggable field? :thinking:

rokaszygmantas commented on src/Adapter/State/CommandHandler/DeleteStateHandler.php:
```diff
@@ -0,0 +1,52 @@
+
+        if (!$this->deleteState($state)) {
+            throw new DeleteStateException(sprintf(
+                    'Cannot delete State object with id "%s".', $state->id)
```
wrong indentation

rokaszygmantas commented on src/Adapter/Zone/ZoneDataProvider.php:
```diff
@@ -0,0 +1,48 @@
+/**
+ * This class will provide data from DB / ORM about Zone
+ */
+class ZoneDataProvider
```
could be `final`

rokaszygmantas commented on src/Core/Domain/State/Query/GetStateForEditing.php:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * @return StateId $stateId
+     */
+    public function getStateId()
```
return type hints can be added everywhere

rokaszygmantas commented on src/Core/Domain/State/ValueObject/StateId.php:
```diff
@@ -0,0 +1,54 @@
+     */
+    public function __construct(int $id)
+    {
+        $this->id = $id;
```
some validation would be nice to have - at least checking if `$id` is a positive integer

rokaszygmantas commented on src/Core/Form/ChoiceProvider/ZoneByIdChoiceProvider.php:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * @param ZoneDataProvider $zoneDataProvider
+     */
+    public function __construct(ZoneDataProvider $zoneDataProvider)
```
we cannot directly depend on adapters in core classes

rokaszygmantas commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -64,4 +78,222 @@ public function getStatesAction(Request $request)
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction($stateId): RedirectResponse
```
missing type hint

rokaszygmantas commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -64,4 +78,222 @@ public function getStatesAction(Request $request)
+    /**
+     * @param Request $request
+     * @param StateFilters $filters
+     *
```
missing `AdminSecurity` annotation

rokaszygmantas commented on src/PrestaShopBundle/Resources/config/routing/admin/improve/international/state.yml:
```diff
@@ -1,6 +1,91 @@
     _controller: 'PrestaShopBundle:Admin\Improve\International\State:getStates'
     _legacy_controller: AdminStates
+
+admin_state_index:
```
route names should be pluralized (`admin_states_...`) everywhere

rokaszygmantas commented on src/PrestaShopBundle/Resources/config/routing/admin/improve/international/state.yml:
```diff
@@ -1,6 +1,91 @@
+
+admin_state_delete:
+  path: /{stateId}/delete
+  methods: [POST, GET]
```
do you need both methods for deletion?

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
CR assets fix

RaimondasSapola committed:
ISO code filtering fix

RaimondasSapola committed:
Missing return types and php docs added

RaimondasSapola committed:
Added missing exception message

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
CR fixes

RaimondasSapola committed:
Code style fixes

RaimondasSapola committed:
removed unneened provider

RaimondasSapola committed:
CR fixes

BlackKerio committed:
After modifying a product by webservice, the product position in the categories changes, even if we will use current position.

BlackKerio committed:
Update classes/Product.php

Co-Authored-By: Pablo Borowicz <pablo.borowicz@prestashop.com>

Commit:
fix: prevent order controller address validation from validating all guest addresses during guest checkout, should only apply to customers

Commit:
fix: coding style update

Commit:
Remove new-theme assets

Commit:
Remove admin default assets

Commit:
Remove themes assets

Commit:
Remove classic theme assets

Commit:
Create make file to install prestashop and build assets with 'make install' command

Commit:
Build assets in travis

BlackKerio committed:
Product.php

After the product has been modified by a webservice, while we also use associations, position in the categories changes

LouiseBonnard committed:
Add Bangladesh localization pack

LouiseBonnard committed:
Add Bosnia and Herzegovina localization pack

LouiseBonnard committed:
Add Cameroon localization pack

LouiseBonnard committed:
Add Senegal localization pack

LouiseBonnard committed:
Add Madagascar localization pack

Commit:
Fixing submenu not showing on classic theme

Commit:
Adjusting quantity on product page if empty or under min qty

tdavidsonas88 committed:
Behat AddRootCategoryHandler

tdavidsonas88 committed:
Behat DeleteCategoryCoverImageHandler

tdavidsonas88 committed:
Behat fixes for EditRootCategory test

Commit:
Remove duplicate var

tdavidsonas88 committed:
Behat DeleteCategoryCoverImage

tdavidsonas88 committed:
cover image variable no longer needed

tdavidsonas88 committed:
Behat DeleteCategoryMenuThumbnailImage

tdavidsonas88 committed:
Behat DeleteCategoryMenuThumbnailImage missing test case

tdavidsonas88 committed:
Behat SetCategoryIsEnabled

tdavidsonas88 committed:
Behat BulkUpdateCategoriesStatusCommand

tdavidsonas88 committed:
Behat GetCategoryIsEnabledHandler

tdavidsonas88 committed:
fixes in code style

tdavidsonas88 committed:
commented out file uploads

tdavidsonas88 committed:
refactored the code after Tomas code review

tdavidsonas88 committed:
code style

jolelievre committed:
Display symbols in form in case of error, remove DefaultLanguage constraint

Commit:
Kick element from callback as he's not used

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

tdavidsonas88 committed:
ps_tmp_img_dir inject through services

tdavidsonas88 committed:
fixed category injection bugg

tdavidsonas88 committed:
fixes for image creation

tdavidsonas88 committed:
possible better ways to solve upload issues - not sure if injection of services has any value now / in the future

tdavidsonas88 committed:
full beauty of the parameters names

tdavidsonas88 committed:
getting temp image dir from the container in test

zuk3975 committed:
Fix suppliers form description to be optional

tdavidsonas88 committed:
refactored paths of images - chosen simpler solution

nesrineabdmouleh committed:
Add numbers for all directories

nesrineabdmouleh committed:
Fix selector id

SimonGrn committed:
Merge pull request #17201 from nesrineabdmouleh/fixEmailThemeSubTabSelector

Functional Tests - Fix "emailThemes/01_previewEmailThemes" test

SimonGrn committed:
Merge pull request #17202 from nesrineabdmouleh/functionaAddNumbersForAllTests

Functional tests - Add numbers for all directories

nesrineabdmouleh committed:
CRUD address

Commit:
Update contributors list for 1.7.6.3

nesrineabdmouleh committed:
Fis selector name

nesrineabdmouleh committed:
Fix identation

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-6

matks committed:
Merge pull request #17221 from matks/merge2develop-6

Merge 1.7.7.x into develop - 17/01/2020

matks committed:
Recompute FOSJSRouting JSON file

jolelievre committed:
Merge pull request #17229 from matks/fix-177x

Recompute FOSJSRouting JSON file

matks committed:
Rebuild Create an Order page assets

matks committed:
Merge pull request #17232 from matks/fix-177x

 Rebuild Create an Order page assets

tdavidsonas88 committed:
PHP_UNIT_FRAMEWORK imported - was missing import after rebase

tdavidsonas88 committed:
using Assert instead of PHPUnit_Framework_Assert.

Commit:
Update configuration

Commit:
PHP cs fixer

Progi1984 committed:
Added RTL Style for Classic Theme

Commit:
Added RTL Style for Classic Theme (#17248)

Added RTL Style for Classic Theme

Progi1984 committed:
Changelog 1.7.6.3

Commit:
Changelog 1.7.6.3 (#17251)

Changelog 1.7.6.3

Commit:
Add Php-cs-fixer into github action

atomiix committed:
Fix deleted products not being displayed in orders

matks committed:
Merge pull request #17246 from PierreRambaud/cs-fixer-update

Update Cs Fixer configuration

matks committed:
Clean a bit files in install folder

matks committed:
Fix some links with http and no https

matthieu-rolland committed:
Merge pull request #17255 from matks/clean-install

Clean a bit files in install folder

tdavidsonas88 committed:
rolled back to $this->configuration->get('_PS_CAT_IMG_DIR_') in handler

matks committed:
Merge pull request #17267 from PrestaShop/1.7.6.x

Merge 1.7.6.x into 1.7.7.x - 21/01/2020

Progi1984 committed:
Merge pull request #17180 from zuk3975/m/supplier/fix-description

Fix suppliers form description to be optional

Progi1984 committed:
Merge pull request #16965 from NeOMakinG/issue10125

Adjusting quantity on product page if empty or under min qty

Progi1984 committed:
Merge pull request #16927 from LouiseBonnard/develop

Add BA, BD, CM, MG and SN localization packs

nesrineabdmouleh committed:
Limit the legal free text length to 50

PierreRambaud committed:
Merge pull request #17257 from matks/fix-https

Fix some links with http and no https

SimonGrn committed:
Merge pull request #17272 from nesrineabdmouleh/fixCheckFooterTextInPDF

Functional tests - Limit the legal free text length to 50

nesrineabdmouleh committed:
Add some fixes

atomiix committed:
Explain data population from OrderDetail

SimonGrn committed:
Merge pull request #17216 from nesrineabdmouleh/functionalCrudAddress

Functional tests - CRUD customer address

nesrineabdmouleh committed:
Address bulk actions

nesrineabdmouleh committed:
Use the same variable addressData

matks committed:
Add new badges to README

nesrineabdmouleh committed:
Fix bulkActionsDeleteButton selector

matks committed:
Enable bulk delete modal for Suppliers, Brands, Brand adresses

matks committed:
Apply php-cs-fixer

SimonGrn committed:
Merge pull request #17281 from nesrineabdmouleh/functionalAddressBulkActions

Functional tests - Addresses bulk actions

Progi1984 committed:
Merge pull request #17043 from jolelievre/fix-new-alternative-currency

Display symbols in form in case of error, remove DefaultLanguage constraint

matks committed:
Merge pull request #17284 from matks/add-badges

Add new badges to README

jolelievre committed:
Display symbols in form in case of error, remove DefaultLanguage constraint

PierreRambaud committed:
Merge pull request #16934 from NeOMakinG/issue16932

Fixing submenu not showing on classic theme

jolelievre committed:
Merge pull request #16670 from PierreRambaud/remove-useless-assets

Remove useless assets

atomiix committed:
Fix deleted products not being displayed in orders (#17254)

Fix deleted products not being displayed in orders

PierreRambaud committed:
Merge pull request #17291 from jolelievre/currency-symbols-error

Display symbols in form in case of error, remove DefaultLanguage constraint

matks committed:
Merge pull request #17286 from matks/enable-bulk-delete-modal-1

Enable bulk delete modal for Suppliers, Brands, Brand adresses

matks committed:
Adapt test for negative amount for money

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-8

matks committed:
Apply php-cs-fixer

matks committed:
Remove asset, they are not committed anymore on develop

jolelievre committed:
Merge pull request #17304 from matks/merge2develop-8

Merge 1.7.7.x into develop - 22/01/2020

tdavidsonas88 committed:
code style fix

Commit:
Build assets for nightly

Commit:
Remove github action to use PrettyCI only

SimonGrn committed:
Merge pull request #17319 from PierreRambaud/fix/nightly-assets

Build assets for nightly

matks committed:
Merge pull request #17018 from tdavidsonas88/feature/category-part-2=integration-tests

Cover Category CommandHandlers and QueryHandlers by behat tests  - Part 2

matks committed:
Merge pull request #17328 from PierreRambaud/fix/prettyci

Remove PHP GitHub Actions and use PrettyCI only

BlackKerio committed:
Merge branch 'develop' into patch-4

BlackKerio committed:
Update Product.php

Progi1984 committed:
Merge pull request #16697 from BlackKerio/patch-4

Fix setWsPositionInCategory and setWsCategories, the product position in the categories changes

Commit:
Prevent order controller from validating all guest addresses (#16620)

Prevent order controller from validating all guest addresses

tdavidsonas88 committed:
implemented the missing controller actions to redirect to Legacy routes

tdavidsonas88 committed:
fixes for styles

tdavidsonas88 committed:
first part of changes after Mathieu review

tdavidsonas88 committed:
fixes for eslint

tdavidsonas88 committed:
refactored to use the object Model toggle status

tdavidsonas88 committed:
changes to avoid duplicate paths

tdavidsonas88 committed:
changes to uniformize status setting and print more specific error

tdavidsonas88 committed:
changes to uniformize status setting and print more specific error
